course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -99,14 +146,129 @@
+        dc.resetForm = function() {
+            dc.formDisplayData = angular.copy(dc.courseInstance);
+        };
 
-        dc.setCourseInstance($routeParams.courseInstanceId);
+        dc.resetFormFromUI = function() {
+            dc.resetForm();
+            dc.resetGlobalAlerts();
+            dc.alerts.global.formReset = {show:true};
+        };
 
+        dc.resetGlobalAlerts = function(scrollToAlerts) {
+            // reposition viewport at top of iframe so we can see
+            // notification messages
+            dc.alerts.global = {};  // reset any existing global alerts
+            // scroll to top of form by default (i.e. if optional scrollToAlerts
+            // argument is not passed in); can disable by setting scrollToAlerts
+            // to false
+            if (scrollToAlerts || scrollToAlerts == null) {
+                scrollTo(0, 0);
+            }
+        };
+
+        dc.submitCourseDetailsForm = function() {
+            dc.courseDetailsUpdateInProgress = true;
+            var postData = {};
+            var url = djangoUrl.reverse(dc.apiProxy,
+                [dc.apiBase + dc.courseInstanceId + '/']);
+            // we could also get these from editable properties on the DOM
+            var fields = [
+                'description',
+                'instructors_display',
+                'location',
+                'meeting_time',
+                'notes',
+                'short_title',
+                'sub_title',
+                'title'
+            ];
+            fields.forEach(function(field) {
+                postData[field] = dc.formDisplayData[field];
+            });
+
+            $http.patch(url, postData)
+                .then(function finalizeCourseDetailsPatch() {
+                    // update Reset button
+                    $.extend(dc.courseInstance, postData);
+                    dc.resetGlobalAlerts();
+                    dc.alerts.global.updateSucceeded = {show: true};
+                }, function cleanUpFailedCourseDetailsPatch(response) {
+                    dc.handleAjaxErrorResponse(response);
+                    dc.resetGlobalAlerts();
+                    dc.alerts.global.updateFailed = {
+                        show: true,
+                        details: response.statusText || 'None'};
+                })
+                .finally( function courseDetailsUpdateNoLongerInProgress() {
+                    dc.courseDetailsUpdateInProgress = false;
+                });
+        };
+
+        dc.init();
+    }
+
+    function editableInputDirective() {
+        return {
+            scope: {
+                editable: '=', // can be < in angular 1.5;
+                               // if not provided, defaults to null/false
+                field: '@',
+                formValue: '=',
+                isLoading: '&',
+                label: '@',
+                maxlength: '@',
+                modelValue: '='
```

Comment:
I'm working on pulling down updated versions of all third-party libraries in `course_info/index.html` to icommons-ui now.